File: app/src/main/java/in/aerem/comconbeacons/UserListItem.java
package in.aerem.comconbeacons;

import in.aerem.comconbeacons.models.UsersResponse;
import org.ocpsoft.prettytime.PrettyTime;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;
import java.util.TimeZone;

public class UserListItem {
    static SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");

    public String username;
    public String location;
    public String time;

    static private String valueOr(String value, String defaultValue) {
        if (value == null || value.isEmpty())
            return defaultValue;
        return value;
    }

    public UserListItem(UsersResponse r) {
        username = valueOr(r.getName(), r.getEmail());
        if (r.getBeacon() == null) {
            location = "None";
        } else {
            location = valueOr(r.getBeacon().getLabel(), r.getBeacon().getBssid());
        }
        time = humanReadableDateInfo(r.getUpdated_at());
    }

    private static String humanReadableDateInfo(String rawDate) {
        format.setTimeZone(TimeZone.getTimeZone("UTC"));
        try {
            PrettyTime p = new PrettyTime(new Locale("ru"));
            Date date = format.parse(rawDate);
            return p.format(date);
        } catch (ParseException e) {
            return "";
        }
    }
}

File: app/src/main/java/in/aerem/comconbeacons/UserListItem.kt
package `in`.aerem.comconbeacons

import `in`.aerem.comconbeacons.models.UserResponse
import java.text.SimpleDateFormat
import java.util.*

class UserListItem {
    var id: Number
    var username: String
    var location: String
    var date: Date

    enum class Status {
        ADVENTURE,
        FREE,
        BUSY,
        UNKNOWN
    }

    lateinit var status: Status

    constructor(r: UserResponse) {
        id = r.id
        username = valueOr(r.name, "Anonymous")
        setStatusFromString(r.status)
        val l = r.location;
        location = l?.label ?: "None"
        date = getDate(r.updated_at)
    }

    fun setStatusFromString(s: String) {
        status =  when (s) {
            "adventure" -> Status.ADVENTURE
            "free" -> Status.FREE
            "busy" -> Status.BUSY
            else -> Status.UNKNOWN
        }
    }

    private var format = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")

    private fun valueOr(value: String?, defaultValue: String): String {
        return if (value == null || value.isEmpty()) defaultValue else value
    }

    private fun getDate(rawDate: String): Date {
        format.timeZone = TimeZone.getTimeZone("Europe/Moscow")
        return format.parse(rawDate)
    }
}
